Add functionality to print metadata

Toreno96 opened the pull request:
Playback status, artist, title, etc.

Decided to use json, as it's easy to parse, especially with tools such as [jq](https://stedolan.github.io/jq/).
Playback status is represented as simple string instead, because I thought it would be an overkill to use json for it, too.

Toreno96 committed:
Add functionality to print metadata

Playback status, artist, title, etc.

icasdri commented on mpris2controller:
```diff
@@ -215,7 +236,12 @@ def is_daemon_up():
+        result = getattr(dbus.SessionBus().get_object(MY_BUS_NAME, MY_PATH), method_name)()
+        if result:
+            try:
+                print(json.dumps(dict(result)))
```
Might be better to check for DBUS's return type (`dbus.Dictionary`, `dbus.Array`, etc.) and display accordingly (instead of trying only for `dict`). I'm assuming `json.dumps` is rejecting the `dbus.Dictionary` directly, otherwise we should definitely just let `json.dumps` deal with it.

Toreno96 replied:
> I'm assuming `json.dumps` is rejecting the `dbus.Dictionary` directly, otherwise we should definitely just let `json.dumps` deal with it.

It is not rejecting it, but then `print(json.dumps(result))` works on `str`/`dbus.String` returned by `Status` method (`ValueError` is not thrown), and prints _quoted_ PlaybackStatus.

You're probably right that it might be better to check for DBUS's return type, but isn't it against common Python practice to not use explicit type-checking (in favor of duck-typing)?

icasdri replied:
Hmm. Good point. I'm not sure I like it either way, and we should probably redesign `call_daemon_method` to be a little less generic, but what you have works for now :P

icasdri commented on mpris2controller:
```diff
@@ -207,6 +208,26 @@ class Controller(dbus.service.Object):
+    @dbus.service.method(dbus_interface=MY_INTERFACE)
+    def Metadata(self):
+        log.info('Method call for Metadata.')
+        metadata = ''
```
Would this return type be consistent with `.get('Metadata')`? I thought the latter was dict-like.

Toreno96 replied:
No, `.get('Metadata')` is of type `dbus.Dictionary(..., signature=dbus.Signature('sv'), variant_level=1)`, while `metadata = ''` is of type `str`.

Now, as you pointed this out, I think it could be more clear to change it to `metadata = None` to represent situation where there is no(ne) metadata because of lack of launched media players.

Or it could be the empty `dbus.Dictionary`.

icasdri replied:
I think `metadata = None` probably makes more sense here then

Toreno96 replied:
I also changed it for the `Status`, and added comments for extra explanation (f55198ff00178a5a5a756ab2ef3cb612d7732fa1).

icasdri commented on mpris2controller:
```diff
@@ -207,6 +208,26 @@ class Controller(dbus.service.Object):
+        log.info('Method call for Status.')
+        status = ''
+        if self.playing:
+            status = Player(self.bus, list(self.playing)[0]).get('PlaybackStatus')
```
`next(iter(self.playing))` seems to be the accepted way of peeking at an element in a set without needing an entirely new copy. So using that might be better.

icasdri commented on mpris2controller:
```diff
@@ -226,7 +252,7 @@ def main():
                         'intelligently controls MPRIS2-compatible media players')
     parser.add_argument('call', nargs='?', metavar='METHOD',
-            help='method to call on daemon (PlayPause, Next, Previous, or Quit)')
+            help='method to call on daemon (PlayPause, Next, Previous, Quit, Status, or Metadata)')
```
Might want to add these to the list at the end of README as well :P

Toreno96 replied:
Sure, I forgot about it :sweat_smile:

icasdri reviewed (changes requested):
This seems like a cool thing we could provide :) (might be useful for e.g. status applets). Was that the use case you had in mind?

Toreno96 commented:
> (might be useful for e.g. status applets). Was that the use case you had in mind?

Yes, it was :) The whole PR originated as local modifications that I made, when I wanted to base my MPRIS XFCE genmon script on the mpris2controller.

Toreno96 committed:
Optimize peeking currently playing player

As specified by @icasdri in
https://github.com/icasdri/mpris2controller/pull/10/files/3418d8d88258e61d8700cf8d35c02de7e212cfd0#r277146040

`next(iter(self.playing))` seems to be the accepted way of peeking at an
element in a set without needing an entirely new copy. So using that
might be better.

Toreno96 committed:
Optimize peeking currently playing player

Forgotten changes, that should be part of
0f07b42af140a4d655758cad350ab925462a5410.

Toreno96 committed:
Change default return type of read properties

As decided in https://github.com/icasdri/mpris2controller/pull/10#discussion_r277195774

Toreno96 committed:
Refactor code duplicates

Toreno96 committed:
Add info about Status and Metadata

Toreno96 commented:
Requested changes done, is everything ok now?

icasdri commented:
Great! Thanks for the contribution :)
